### PR TITLE
DOC-614 | Added `readOwnWrites` option for UPSERT operations

### PR DESCRIPTION
**Pull request by nerpaula**
### Description

Added `readOwnWrites` option for UPSERT operations.

#### Upstream PRs

- 3.10:
- 3.11:
- 3.12: https://github.com/arangodb/arangodb/pull/20134

### CODE REVIEW

File: site/content/3.12/aql/high-level-operations/upsert.md [hunk 1]
```diff
@@ -243,6 +243,20 @@ UPSERT { a: 1234 }
...
+The `readOwnWrites` option allows `UPSERT` operations to process its inputs one
+by one. The default value is `true`. When enabled, the `UPSERT` operation can
+observe its own writes and can handle modifying the same target document multiple
+times in the same query.
+
+When the option is set to `false`, an `UPSERT` operation processes its inputs
+in batches. A batch has normally 1000 inputs, so it can lead to a faster execution.
+However, when using batches, the `UPSERT` operation cannot observe its own writes.
+The `readOwnWrites` option should be set to `false` only if it can
+be guaranteed that the input of the `UPSERT` leads to disjoint documents being
```
Comment (Simran-B) [inline site/content/3.12/aql/high-level-operations/upsert.md:257]:
```suggestion
You should only set the `readOwnWrites` option to `false` if you can
guarantee that the input of the `UPSERT` leads to disjoint documents being
```

File: site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md [hunk 2]
```diff
@@ -174,6 +174,20 @@ UPDATE { logins: OLD.logins + 1 } IN users
 
 Read more about [`UPSERT` operations](../../aql/high-level-operations/upsert.md) in AQL.
 
+### `readOwnWrites` option for `UPSERT` operations
+
+The `readOwnWrites` option has been added for `UPSERT` operations. The default
```
Comment (Simran-B) [inline site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md:179]:
```suggestion
A `readOwnWrites` option has been added for `UPSERT` operations. The default
```

File: site/content/3.12/aql/high-level-operations/upsert.md [hunk 3]
```diff
@@ -243,6 +243,20 @@ UPSERT { a: 1234 }
...
 
+### `readOwnWrites`
+
+The `readOwnWrites` option allows `UPSERT` operations to process its inputs one
+by one. The default value is `true`. When enabled, the `UPSERT` operation can
+observe its own writes and can handle modifying the same target document multiple
+times in the same query.
+
+When the option is set to `false`, an `UPSERT` operation processes its inputs
+in batches. A batch has normally 1000 inputs, so it can lead to a faster execution.
```
Comment (Simran-B) [inline site/content/3.12/aql/high-level-operations/upsert.md:254]:
```suggestion
in batches. Normally, a batch has 1000 inputs, which can lead to a faster execution.
```

File: site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md [hunk 4]
```diff
@@ -174,6 +174,20 @@ UPDATE { logins: OLD.logins + 1 } IN users
...
+value is `true`. When set to `true`, an `UPSERT` operation processes its
+inputs one by one. This way, the operation can observe its own writes and can
+handle modifying the same target document multiple times in the same query.
+
+When the option is set to `false`, an `UPSERT` operation processes its inputs
+in batches. A batch has normally 1000 inputs, so it can lead to a faster execution.
```
Comment (Simran-B) [inline site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md:185]:
```suggestion
in batches. Normally, a batch has 1000 inputs, which can lead to a faster execution.
```

File: site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md [hunk 5]
```diff
@@ -174,6 +174,20 @@ UPDATE { logins: OLD.logins + 1 } IN users
...
+However, when using batches, the `UPSERT` operation cannot observe its own writes.
+Therefore, the `readOwnWrites` option should be set to `false` only if it can
+be guaranteed that the input of the `UPSERT` leads to disjoint documents being
```
Comment (Simran-B) [inline site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md:188]:
```suggestion
Therefore, you should only set the `readOwnWrites` option to `false` if you can
guarantee that the input of the `UPSERT` leads to disjoint documents being
```

File: site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md [hunk 6]
```diff
@@ -174,6 +174,20 @@ UPDATE { logins: OLD.logins + 1 } IN users
 
 Read more about [`UPSERT` operations](../../aql/high-level-operations/upsert.md) in AQL.
 
+### `readOwnWrites` option for `UPSERT` operations
+
+The `readOwnWrites` option has been added for `UPSERT` operations. The default
+value is `true`. When set to `true`, an `UPSERT` operation processes its
```
Comment (Simran-B) [inline site/content/3.12/release-notes/version-3.12/whats-new-in-3-12.md:180]:
```suggestion
value is `true` and the behavior is identical to previous versions of ArangoDB that
do not have this option. When enabled, an `UPSERT` operation processes its
```

**Review by Simran-B — CHANGES_REQUESTED** [review]

File: site/content/3.12/aql/high-level-operations/upsert.md [hunk 7]
```diff
@@ -243,6 +243,20 @@ UPSERT { a: 1234 }
...
+However, when using batches, the `UPSERT` operation cannot observe its own writes.
```
[the remark this answers is not in the document]
Comment (nerpaula) [reply site/content/3.12/aql/high-level-operations/upsert.md:255]: How would you rephrase it?

[the remark this answers is not in the document]
Comment (Simran-B) [reply site/content/3.12/aql/high-level-operations/upsert.md:255]: However, when using batches, the `UPSERT` operation can essentially not observe its own writes.

File: site/content/3.12/aql/high-level-operations/upsert.md [hunk 8]
```diff
@@ -243,6 +243,20 @@ UPSERT { a: 1234 }
...
+### `readOwnWrites`
+
+The `readOwnWrites` option allows an `UPSERT` operation to process its inputs one
+by one. The default value is `true`. When enabled, the `UPSERT` operation can
+observe its own writes and can handle modifying the same target document multiple
+times in the same query.
+
+When the option is set to `false`, an `UPSERT` operation processes its inputs
+in batches. Normally, a batch has 1000 inputs, which can lead to a faster execution.
+However, when using batches, the `UPSERT` operation cannot observe its own writes.
```
Comment (Simran-B) [inline site/content/3.12/aql/high-level-operations/upsert.md:255]:
```suggestion
However, when using batches, the `UPSERT` operation can essentially not observe its own writes.
```

**Review by Simran-B — APPROVED** [review]

**Pull request merged**